src/Servers/Kestrel/Core/src/Internal/Http/HttpResponseHeaders.cs

```diff
@@ -30,19 +30,27 @@ public Enumerator GetEnumerator()
         internal void CopyTo(ref BufferWriter<PipeWriter> buffer)
         {
             CopyToFast(ref buffer);
-            if (MaybeUnknown != null)
+
+            var extraHeaders = MaybeUnknown;
```

Comment:
Pardon my ignorance, but why is it better to check a local variable here rather than the property?

Reply:
As we are using it 3 times (`is null`, `.Count`, `.GetEnumerator`) while the Jit *may* read it once into a register; its just to encourage that to happen and it use a CPU register rather than main memory for the reference.

We then pass it as a parameter to `CopyExtraHeaders` and make that method static, so it keeps it in a register rather than looking it up again from main memory (e.g. `this.MaybeUnknown`).

Reply:
Note if there was a lot of stack and registers in use by the method (as there were previously) then this wouldn't be an optimization as it would just copy the value to stack instead (as there would be no registers free)